addons/crm/models/crm_lead.py

```diff
@@ -537,13 +537,18 @@ def _compute_recurring_revenue_prorated(self):
         for lead in self:
             lead.recurring_revenue_prorated = (lead.recurring_revenue or 0.0) * (lead.probability or 0) / 100.0
 
-    def _compute_calendar_event_count(self):
-        meeting_data = self.env['calendar.event'].sudo()._read_group([
-            ('opportunity_id', 'in', self.ids)
-        ], ['opportunity_id'], ['__count'])
-        mapped_data = {opportunity.id: count for opportunity, count in meeting_data}
-        for lead in self:
-            lead.calendar_event_count = mapped_data.get(lead.id, 0)
+    @api.depends('calendar_event_ids', 'calendar_event_ids.start')
+    def _compute_meeting_display_date(self):
+        now = fields.Datetime.now()
+        leads_with_meetings = self.filtered('calendar_event_ids')
+        (self - leads_with_meetings).meeting_display_date = False
+        for lead in leads_with_meetings:
+            start_dts = lead.calendar_event_ids.mapped('start')
+            future_start_dts = [dt for dt in start_dts if dt > now]
+            if future_start_dts:
+                lead.meeting_display_date = min(future_start_dts)
+            else:
+                lead.meeting_display_date = max(start_dts)
```

Comment:
We could still use a read_group here:
```suggestion
        now = fields.Datetime.now()
        meeting_data = self.env['calendar.event'].sudo()._read_group([
            ('opportunity_id', 'in', self.ids),
        ], ['opportunity_id'], ['start:array_agg', 'start:max'])
        mapped_data = {
            lead: {
                'last_meeting_date': last_meeting_date,
                'next_meeting_date': min([dt for dt in meeting_start_dates if dt > now] or [False]),
            } for lead, meeting_start_dates, last_meeting_date in meeting_data
        }
        for lead in self:
            lead_meeting_info = mapped_data.get(lead)
            if not lead_meeting_info:
                lead.meeting_display_date = False
            elif lead_meeting_info['next_meeting_date']:
                lead.meeting_display_date = lead_meeting_info['next_meeting_date']
            else:
                lead.meeting_display_date = lead_meeting_info['last_meeting_date']
```